### PR TITLE
Initial Byron isolation 1 of n

### PR DESCRIPTION
**Pull request by Jimbo4350**
# Changelog

```yaml
- description: |
    Collateral inputs and the minimum value calculation are not available in the Byron era. Update the code to reflect this.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Additional context for the PR goes here. If the PR fixes a particular issue please provide a [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=) to the issue.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs [hunk 1]
```diff
@@ -1023,14 +1023,14 @@ runTransactionCalculateMinValueCmd
       , txOut
       } = do
   pp <- firstExceptT TxCmdProtocolParamsError (readProtocolParameters protocolParamsFile)
-  out <- toTxOutInAnyEra eon txOut
+  out <- toTxOutInShelleyBasedEra eon txOut
   -- TODO: shouldn't we just require shelley based era here instead of error-ing for byron?
-  forEraInEon eon (error "runTransactionCalculateMinValueCmd: Byron era not implemented yet") $ \sbe -> do
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs:1028]: Byron was never supported

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Transaction.hs [hunk 2]
```diff
@@ -78,12 +78,17 @@ pTransactionCmds era envCli =
         $ subParser "calculate-min-fee"
         $ Opt.info (pTransactionCalculateMinFee envCli)
         $ Opt.progDesc "Calculate the minimum fee for a transaction."
-    , Just
-        $ subParser "calculate-min-required-utxo"
-        $ Opt.info (pTransactionCalculateMinReqUTxO era)
-        $ Opt.progDesc "Calculate the minimum required UTxO for a transaction output."
-    , Just
-        $ pCalculateMinRequiredUtxoBackwardCompatible era
+    , caseByronOrShelleyBasedEra
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Transaction.hs:81]: The goal is to not need `caseByronOrShelleyBasedEra` in the future. Required byron based functionality will be hard coded and available under the `byron` command group.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs:1028]: `TODO` can be removed now

**Review by carbolymer — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Types/Common.hs [hunk 3]
```diff
@@ -389,6 +390,14 @@ instance ToJSON SlotsTillKesKeyExpiry where
...
   parseJSON v = SlotsTillKesKeyExpiry <$> parseJSON v
 
+
+data TxOutShelleyBasedEra
+  = TxOutShelleyBasedEra
+      !(Address ShelleyAddr)
+      Value
+      TxOutDatumAnyEra
+      ReferenceScriptAnyEra
+  deriving Show
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/Types/Common.hs:393]: We should exclude `ShelleyBasedEra` from all type and function names and have all the Byron specific types and functions of carr the burden of qualifying that it is only valid for `ByronOnly`.

Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Types/Common.hs:393]: Yep :+1: . This naming is just to make it painfully obvious what I'm doing. Eventually these types won't have anything in its name to indicate the era.

**Comment by Jimbo4350** [comment]
Subsumed by #463

**Pull request closed without merging**